### PR TITLE
Set context before calling use_sudo_password()

### PR DESCRIPTION
**Pull request by jaysoffian**
Set context before calling `use_sudo_password()` in case it wants access
to host data. A use case is setting the sudo password in group data
files, and then later providing it via config.USE_SUDO_PASSWORD. e.g.

    config.USE_SUDO_PASSWORD = lambda: host.data.sudo_password

### CODE REVIEW

File: pyinfra/connectors/util.py [hunk 1]
```diff
@@ -179,7 +180,9 @@ def _get_sudo_password(host, use_sudo_password):
             host.connector_data["sudo_password"] = sudo_password
         sudo_password = sudo_password
     elif callable(use_sudo_password):
-        sudo_password = use_sudo_password()
+        with ctx_state.use(state):
```
Comment (Fizzadar) [inline pyinfra/connectors/util.py:183]: This shouldn't be required as the state context is set for the entire run @ https://github.com/Fizzadar/pyinfra/blob/5fedec46b127d16d59ca73537f90dadff647511d/pyinfra_cli/main.py#L380

**Review by Fizzadar — COMMENTED** [review]
Thank you for the PR @jaysoffian, would be good to have this working!

I'm not sure about setting the host context here though, might be better set for the entire duration of operation execution; I just pushed up https://github.com/Fizzadar/pyinfra/pull/869 to test if this works as expected...

**Comment by Fizzadar** [comment]
@jaysoffian I have just merged in https://github.com/Fizzadar/pyinfra/commit/591ea61e3b29210ccb5ac71665920f1374a51e18 which I believe solves this issue (and any other host state needs within operation execution context).

**Comment by jaysoffian** [comment]
> @jaysoffian I have just merged in [591ea61](https://github.com/Fizzadar/pyinfra/commit/591ea61e3b29210ccb5ac71665920f1374a51e18) which I believe solves this issue (and any other host state needs within operation execution context).

Indeed. Thanks!

**Pull request closed without merging**